docs/builds.rst

```diff
@@ -0,0 +1,328 @@
+additional options such as pulling Git submodules and setting additional environment variables for the build process.
+
+You can find information regarding the ``revisions`` key on the :ref:`Revisions` page.
+
+Stating a Git Repository
+------------------------
+
+To state our git repository, we use the keys
+
+- ``name``: (arbitrary) name of the build
+- ``git``: link to the git-repository
+
+as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to state a git-repository in the experiments.yml file.
+
+    builds:
+      - name: simexpal
+        git: https://github.com/hu-macsy/simexpal
+
+.. _PullingGitSubmodules:
+
+Pulling Git Submodules
+----------------------
+
+If your program contains Git submodules you need to set the
+
+- ``recursive-clone``: boolean (``True``/``False``) - whether to pull git submodules recursively or not
+
+key to ``True`` in order for simexpal to pull the respective files.
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to let simexpal pull Git submodules in the experiments.yml file.
+
+    builds:
+      - name: networkit
+        git: https://github.com/networkit/networkit
+        recursive-clone: True
+
+If ``recursive-clone`` is not set, it will default to ``False``.
+
+Automated Builds
+----------------
+
+To enable automated build support, we can specify the following keys:
+
+- ``regenerate``: list of dictionaries containing regeneration parameters
+- ``configure``: list of dictionaries containing configuration parameters
+- ``compile``: list of dictionaries containing compilation parameters
+- ``install``: list of dictionaries containing installation parameters
+
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to specify configuration, compilation and installation parameters in the experiments.yml file.
+
+   builds:
+     - name: simexpal
+       git: 'https://github.com/hu-macsy/simexpal'
+       configure:
+         - args:
+             - 'cmake'
+             - '-DCMAKE_INSTALL_PREFIX=@THIS_PREFIX_DIR@'
+             - '@THIS_CLONE_DIR@/examples/sorting_cpp/'
+       compile:
+         - args:
+             - 'make'
+             - '-j@PARALLELISM@'
+       install:
+         - args:
+             - 'make'
+             - 'install'
+
+To specify the build parameters, we will use the ``args`` key and set the value to a list of arguments. Arguments
+are stated separately e.g. ``make install`` becomes a list containing ``make`` and ``install``. In the example
+above, we used CMake as build system; however, simexpal is independent of the particular build system in use.
+
+The ``regenerate`` key works analogously to the other keys and its purpose is to execute commands between pulling
+and configuring the project. An example would be, to pull submodules recursively via
+``git submodule update --init --recursive`` (this can actually be done by setting another option: see
+:ref:`PullingGitSubmodules`).
```

Comment:
Therefore, it is not a good example ;)

Reply:
I removed this example and used the one you mentioned above.